src/Dapr.Workflow/DaprWorkflowClient.cs

```diff
@@ -0,0 +1,310 @@
+        /// </summary>
+        /// <param name="instanceId">The instance ID of the workflow to resume.</param>
+        /// <param name="reason">The optional resume reason.</param>
+        /// <param name="cancellation">
+        /// A <see cref="CancellationToken"/> that can be used to cancel the resume operation. Note, cancelling this token
+        /// does <b>not</b> re-suspend the workflow if resume was successful.
+        /// </param>
+        /// <returns>A task that completes when the resume has been committed to the backend.</returns>
+        public Task ResumeWorkflowAsync(
+            string instanceId,
+            string? reason = null,
+            CancellationToken cancellation = default)
+        {
+            return this.innerClient.ResumeInstanceAsync(instanceId, reason, cancellation);
+        }
+
+        /// <summary>
+        /// Purges workflow instance state from the workflow state store.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This method can be used to permanently delete workflow metadata from the underlying state store,
+        /// including any stored inputs, outputs, and workflow history records. This is often useful for implementing
+        /// data retention policies and for keeping storage costs minimal. Only workflow instances in the
+        /// <see cref="WorkflowRuntimeStatus.Completed"/>, <see cref="WorkflowRuntimeStatus.Failed"/>, or
+        /// <see cref="WorkflowRuntimeStatus.Terminated"/> state can be purged.
+        /// </para>
+        /// </remarks>
+        /// <param name="instanceId">The unique ID of the workflow instance to purge.</param>
+        /// <param name="cancellation">
+        /// A <see cref="CancellationToken"/> that can be used to cancel the purge operation.
+        /// </param>
+        /// <returns>
+        /// Returns a task that completes when the purge operation has completed. The value of this task will be
+        /// <c>true</c> if the workflow state was found and purged successfully; <c>false</c> otherwise.
+        /// </returns>
+        public async Task<bool> PurgeInstanceAsync(string instanceId, CancellationToken cancellation = default)
+        {
+            PurgeResult result = await this.innerClient.PurgeInstanceAsync(instanceId, cancellation);
+            return result.PurgedInstanceCount > 0;
        }
+
+        /// <summary>
+        /// Disposes any unmanaged resources associated with this client.
+        /// </summary>
+        public ValueTask DisposeAsync()
+        {
+            return ((IAsyncDisposable)this.innerClient).DisposeAsync();
+        }
+    }
+
+    /// <summary>
+    /// Deprecated. Use <see cref="DaprWorkflowClient"/> instead.
+    /// </summary>
+    [Obsolete($"Deprecated. Use {nameof(DaprWorkflowClient)} instead.")]
+    public sealed class WorkflowEngineClient : DaprWorkflowClient

```

Comment:
Does moving the class to a different file cause any weird binary breaking changes? I appreciate the less code of removing the class and putting it here, just want to make sure we don't have any weirdness that happens.

Reply:
Moving to a different file doesn't have any impact on the built binaries. But I'll go ahead and move this to its own file anyways to avoid confusion.